### PR TITLE
[Tests-Only] Api test for copying file within a public link folder

### PR DESCRIPTION
**Pull request by swoichha**
## Description
API test scenario for copying a file within within a public link folder

## Related Issue
https://github.com/owncloud/ocis-reva/issues/310

## How Has This Been Tested?
- Test in ocis: https://github.com/owncloud/ocis/pull/358
- Test in ocis-reva: https://github.com/owncloud/ocis-reva/pull/334

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    And as "Alice" file "/PARENT/testfile.txt" should exist
+    And as "Alice" file "/PARENT/copy1.txt" should exist
+
+  Scenario: Copy folder within a public link folder new public WebDAV API
+    Given the administrator has enabled DAV tech_preview
+    And user "Alice" has created folder "/PARENT/testFolder"
+    And user "Alice" has created a public link share with settings
+      | path        | /PARENT                   |
+      | permissions | read,update,create,delete |
+    When user "Alice" copies file "/testFolder" to "/testFolder-copy" using the new public WebDAV API
```
Comment (jasson99) [inline tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature:25]:
```suggestion
    When user "Alice" copies folder "/testFolder" to "/testFolder-copy" using the new public WebDAV API
```

File: tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    Then the HTTP status code should be "201"
+    And as "Alice" file "/PARENT/testFolder" should exist
```
Comment (jasson99) [inline tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature:27]:
```suggestion
    And as "Alice" folder "/PARENT/testFolder" should exist
```

File: tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    And as "Alice" file "/PARENT/testFolder-copy" should exist
```
Comment (jasson99) [inline tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature:28]:
```suggestion
    And as "Alice" folder "/PARENT/testFolder-copy" should exist
```

File: tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  Scenario: Copy file within a public link folder new public WebDAV API
+    Given the administrator has enabled DAV tech_preview
+    And user "Alice" has uploaded file with content "some data" to "/PARENT/testfile.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | /PARENT                   |
+      | permissions | read,update,create,delete |
+    When user "Alice" copies file "/testfile.txt" to "/copy1.txt" using the new public WebDAV API
+    Then the HTTP status code should be "201"
+    And as "Alice" file "/PARENT/testfile.txt" should exist
+    And as "Alice" file "/PARENT/copy1.txt" should exist
```
Comment (individual-it) [inline tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature:17]: please also check the content of the copy and the original file, just to make sure the data hasn't been corrupted

File: tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  Scenario: Copy folder within a public link folder new public WebDAV API
+    Given the administrator has enabled DAV tech_preview
+    And user "Alice" has created folder "/PARENT/testFolder"
```
Comment (individual-it) [inline tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature:21]: place some file in the folder, so the copy operation is more than just creating a folder

File: tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature [hunk 6]
```diff
@@ -0,0 +1,40 @@
+@api @files_sharing-app-required @public_link_share-feature-required @skipOnOcis @issue-ocis-reva-310
```
Comment (individual-it) [inline tests/acceptance/features/apiSharePublicLink2/copyFromPublicLink.feature:1]: I believe that should work on OCIS now, so no need to skip it on ocis

**Review by individual-it — CHANGES_REQUESTED** [review]
more tests should be added to cover copy operations that should not work, but that can be in a new PR

**Review by individual-it — APPROVED** [review]
please squash before merging!

**Review by jasson99 — COMMENTED** [review]
LGTM

**Pull request merged**